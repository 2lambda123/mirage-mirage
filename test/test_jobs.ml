let context_singleton key value =
  let info = Cmdliner.Term.info "" in
  let term =
    Functoria_key.context
      ~with_required:false
      (Functoria_key.Set.singleton @@ Mirage_key.abstract key)
  in
  let argv = [|"program"; "--target"; value|] in
  match Cmdliner.Term.eval ~argv (term, info) with
  | `Ok x -> x
  | _ -> assert false

let print_banner s =
  print_endline s;
  print_endline @@ String.make (String.length s) '=';
  print_newline ()

let test target =
  print_banner target;
  let context =
    context_singleton
    Mirage_key.target
    target
  in
  Mirage_job.dry_run_trace ~files:[] @@
  Mirage.configure @@
  Functoria.Info.create
    ~packages:[]
    ~keys:[]
    ~context
    ~name:"NAME"
    ~build_dir:(Fpath.v "BUILD_DIR");
  print_newline ()

let () =
  List.iter
    test
    [ "unix"
    ; "xen"
    ; "virtio"
    ]
  
