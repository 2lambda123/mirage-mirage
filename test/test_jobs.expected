unix
====

Exists? myocamlbuild.ml -> false
Write to myocamlbuild.ml (0 bytes)
Write (fmt) to mirage-unikernel-NAME-unix.opam (mode: default, purpose: opam file)
Write (fmt) to Makefile (mode: default, purpose: Makefile)

xen
===

Exists? myocamlbuild.ml -> false
Write to myocamlbuild.ml (0 bytes)
Write (fmt) to mirage-unikernel-NAME-xen.opam (mode: default, purpose: opam file)
Write (fmt) to Makefile (mode: default, purpose: Makefile)
Write (fmt) to NAME.xl (mode: default, purpose: xl file)
Write (fmt) to NAME.xl.in (mode: default, purpose: xl file)
Write (fmt) to NAME.xe (mode: 0755, purpose: xe file)
Write (fmt) to NAME_libvirt.xml (mode: default, purpose: libvirt.xml)

virtio
======

Exists? myocamlbuild.ml -> false
Write to myocamlbuild.ml (0 bytes)
Write (fmt) to mirage-unikernel-NAME-virtio.opam (mode: default, purpose: opam file)
Write (fmt) to Makefile (mode: default, purpose: Makefile)
Write (fmt) to NAME_libvirt.xml (mode: default, purpose: libvirt.xml)

